config/configuration.go

```diff
@@ -137,19 +138,30 @@ func init() {
 //GetCORS gets CORS related config
 func GetCORS() cors.Config {
 	viper.SetDefault("cors.AllowAllOrigins", true)
-	viper.SetDefault("cors.AllowOrigins", []string{"http://", "https://"})
+	viper.SetDefault("cors.AllowOrigins", []string{})
+	viper.SetDefault("cors.AllowOriginsRegexp", "")
 	viper.SetDefault("cors.AllowMethods", []string{"PUT", "DELETE", "GET", "POST", "OPTIONS"})
 	viper.SetDefault("cors.AllowHeaders", []string{"Origin", "Authorization", "Content-Type", "secretId"})
 	viper.SetDefault("cors.ExposeHeaders", []string{"Content-Length"})
 	viper.SetDefault("cors.AllowCredentials", true)
 	viper.SetDefault("cors.MaxAge", 12)
 
 	config := cors.DefaultConfig()
-	cors.DefaultConfig()
 	config.AllowAllOrigins = viper.GetBool("cors.AllowAllOrigins")
 	if !config.AllowAllOrigins {
+		allowOriginsRegexp := viper.GetString("cors.AllowOriginsRegexp")
+		if allowOriginsRegexp != "" {
+			originsRegexp, err := regexp.Compile(allowOriginsRegexp)
+			if err == nil {
+				config.AllowOriginFunc = func(origin string) bool {
+					return originsRegexp.Match([]byte(origin))
```

Comment:
I would highlight somewhere in the code that the regexp is matched as substring pattern, which is not that usual for similar config options.